numpy/_core/_add_newdocs.py

```diff
@@ -6290,7 +6290,27 @@
     """))
 
 add_newdoc('numpy._core.multiarray', 'dtype', ('type',
-    """The type object used to instantiate a scalar of this data-type."""))
+    """
+    The type object used to instantiate a scalar of this data-type.
+
+    For specific data types such as numpy.int32, numpy.float64, this attribute provides
+    the Python class corresponding to the scalar values of the array elements.
+
+    For the base class `numpy.dtype` and certain abstract `dtype` types,
+    this attribute is `None`.
```

Comment:
Quick question: 
This is only the case for class attributes, and not for instance attributes, right? Because otherwise, [the `dtype.type` stubs](https://github.com/numpy/numpy/blob/dcbc24e9b5872fe75c845d2f9e1a96bd0200e9f5/numpy/__init__.pyi#L1581-L1582) are wrong 😅